### PR TITLE
Allow install without thrust

### PR DESCRIPTION
**Pull request by niboshi**
It takes so much time to build cupy from source with thrust enabled, which is not needed in most daily development.

With this PR, thrust can be disabled by setting an environment variable `CUPY_SETUP_ENABLE_THRUST=0`.

Build time:
* before: 9m17s
* thrust disabled: 2m49s

### CODE REVIEW

**Comment by niboshi** [comment]
I believe we don't have to care about failed tests, because when setting this environment variable, the developer should know what he's doing.

**Comment by emcastillo** [comment]
Perfect, just wanted to double check! 😀

**Comment by emcastillo** [comment]
Jenkins, test this please

File: cupy_setup_build.py [hunk 1]
```diff
@@ -198,6 +184,24 @@
...
+            'thrust/sort.h',
+        ],
+        'libraries': [
+            'cudart',
+        ],
+        'check_method': build.check_cuda_version,
+    })
+
+
 def convert_modules_for_hip():
```
Comment (leofang) [inline cupy_setup_build.py]: I guess you need to handle this function `convert_modules_for_hip()` when `CUPY_SETUP_ENABLE_THRUST=0`?

Comment (niboshi) [reply cupy_setup_build.py]: Thank you for pointing out!

@okuta 
What are these lines?

https://github.com/cupy/cupy/blob/bcf2ec3f09c31042c6e413a7a160fd92f7a3f215/cupy_setup_build.py#L203-L204

File: cupy_setup_build.py [hunk 2]
```diff
@@ -198,6 +184,24 @@
 ]
 
 
+if 1 == int(os.environ.get('CUPY_SETUP_ENABLE_THRUST', 1)):
```
Comment (okuta) [inline cupy_setup_build.py]:
```suggestion
if bool(int(os.environ.get('CUPY_SETUP_ENABLE_THRUST', 1))):
```

File: cupy_setup_build.py (hunk 1 above)
Comment (okuta) [reply cupy_setup_build.py]: `convert_modules_for_hip` is called some times.

Comment (niboshi) [reply cupy_setup_build.py]: With different number of elements each time?
What's does the magic number 3 mean?

Comment (niboshi) [reply cupy_setup_build.py]: OK, it looks 3 is the number of modules `MODULE` will have after this function is called for the first time. (`cuda`, `cusolver` and `thrust`, I guess?)

I think I need to rewrite this function.

Comment (niboshi) [reply cupy_setup_build.py]: 0836f9cfe25061ad2108a456b01916b56eedb04b

**Comment by niboshi** [comment]
PTAL

**Comment by emcastillo** [comment]
Do we have any means of testing this in an AMD gpu?

**Comment by leofang** [comment]
> Do we have any means of testing this in an AMD gpu?

This is important because we often break HIP support silently, and CI seems unable to catch it. The HIP support just got fixed in #2737 with a user reported a successful AMD build.

**Comment by emcastillo** [comment]
I don't feel confident enough to assert that this changeset will not break ROCm support without testing it so I just want to be cautious. If we can test it somehow it should be great. Otherwise, I think that we should merge this after the release.

**Comment by emcastillo** [comment]
Jenkins, test this please

**Comment by niboshi** [comment]
Conflict resolved.

**Comment by emcastillo** [comment]
Jenkins, test this please

**Pull request merged**